src/domain_bridge/domain_bridge.cpp

```diff
@@ -107,41 +108,33 @@ class DomainBridgeImpl
       type, "rosidl_typesupport_cpp");
   }
 
-  std::shared_ptr<GenericPublisher> create_publisher(
+  std::shared_ptr<rclcpp::GenericPublisher> create_publisher(
     rclcpp::Node::SharedPtr node,
     const std::string & topic_name,
+    const std::string & type,
     const rclcpp::QoS & qos,
-    const rosidl_message_type_support_t & typesupport_handle,
     rclcpp::CallbackGroup::SharedPtr group)
   {
-    auto publisher = std::make_shared<GenericPublisher>(
-      node->get_node_base_interface().get(),
-      typesupport_handle,
-      topic_name,
-      qos);
+    auto publisher = node->create_generic_publisher(topic_name, type, qos);
     node->get_node_topics_interface()->add_publisher(publisher, std::move(group));
```

Comment:
you can delete this line, `create_generic_publisher()` already does this [here](https://github.com/ros2/rclcpp/blob/1037822a63330495dcf0de5e8f20544375a5f116/rclcpp/include/rclcpp/create_generic_publisher.hpp#L63).

IIRC, you can pass a callback group in the [options](https://github.com/ros2/rclcpp/blob/1037822a63330495dcf0de5e8f20544375a5f116/rclcpp/include/rclcpp/node.hpp#L290) argument.

Reply:
I've removed this line and passed the callbacks into the options argument.